In [1]:
import pandas as pd
import sqlite3

In [4]:
path = 'C:/my_workspace/Github/MentalHealth_SQL_Project/data/'
health = pd.read_csv(path + 'mental_health.csv')

In [5]:
conn = sqlite3.connect('PandoraDB.db')
health.to_sql('health', conn, index=False, if_exists='replace')

286

*Contando o tamanho dos dados.*

In [6]:
query = \
"SELECT COUNT(*) AS total_records \
FROM health"

results = pd.read_sql_query(query, conn)
results

total_records
0            286

*Selecionando tudo para analisar as colunas.*

In [8]:
query = \
"SELECT * \
FROM health \
LIMIT 5;"

results = pd.read_sql_query(query, conn)
results

inter_dom Region  Gender Academic   Age  Age_cate  Stay Stay_Cate  Japanese  \
0     Inter    SEA    Male     Grad  24.0       4.0   5.0      Long       3.0   
1     Inter    SEA    Male     Grad  28.0       5.0   1.0     Short       4.0   
2     Inter    SEA    Male     Grad  25.0       4.0   6.0      Long       4.0   
3     Inter     EA  Female     Grad  29.0       5.0   1.0     Short       2.0   
4     Inter     EA  Female     Grad  28.0       5.0   1.0     Short       1.0   

  Japanese_cate  ...  Friends_bi Parents_bi Relative_bi Professional_bi  \
0       Average  ...         Yes        Yes          No              No   
1          High  ...         Yes        Yes          No              No   
2          High  ...          No         No          No              No   
3           Low  ...         Yes        Yes         Yes             Yes   
4           Low  ...         Yes        Yes          No             Yes   

  Phone_bi Doctor_bi religion_bi  Alone_bi Others_bi  Internet_bi  
0       No        No          No        No        No           No  
1       No        No          No        No        No           No  
2       No        No          No        No        No           No  
3       No        No          No        No        No           No  
4       No       Yes         Yes        No        No           No  

[5 rows x 50 columns]

*Temos 50 colunas.*

*Vamos olhar nossa primeira coluna, inter_dom.*

In [9]:
query = \
"SELECT DISTINCT (inter_dom) \
FROM health;"

results = pd.read_sql_query(query, conn)
results

inter_dom
0     Inter
1       Dom
2      None

*Temos 3 valores distintos. International, Domestic and None.*

*Vamos olhar a distribuição.*

In [10]:
query = \
"SELECT inter_dom, COUNT(inter_dom) AS inter_dom_count \
FROM health \
GROUP BY inter_dom;"

results = pd.read_sql_query(query, conn)
results

inter_dom  inter_dom_count
0      None                0
1       Dom               67
2     Inter              201

*Vamos olhar quantos valores None temos.*

*Domestic 67 e International são 201 estudantes.*

In [13]:
query = "SELECT Gender, Age \
        FROM health \
        WHERE inter_dom = 'Inter' \
        GROUP BY Gender \
        ORDER BY Age DESC;"

results = pd.read_sql_query(query, conn)
results

Gender   Age
0  Female  29.0
1    Male  24.0

In [12]:
query = \
"SELECT Gender, AVG(Age) AS Avg, COUNT (Age) AS count \
FROM health \
WHERE inter_dom = 'Inter' \
GROUP BY Gender \
ORDER BY Age DESC;"

results = pd.read_sql_query(query, conn)
results

Gender        Avg  count
0  Female  20.789062    128
1    Male  21.452055     73